src/main/java/core/basesyntax/Rectangle.java

```diff
@@ -27,4 +27,37 @@ public Integer getLength() {
     public String getColor() {
         return color;
     }
+
+    @Override
+    public int hashCode() {
+        int result = 17;
+        final int prime = 31;
+        result = prime * result + (width != null ? width.hashCode() : 0);
+        result = prime * result + (length != null ? length.hashCode() : 0);
+        result = prime * result + (color != null ? color.hashCode() : 0);
+        ;
+        return result;
+    }
+
+    @Override
+    public boolean equals(Object obj) {
+        if (this == obj) {
+            return true;
+        }
+
+        if (obj == null) {
+            return false;
+        }
+
+        if (obj instanceof Rectangle) {
```

Comment:
what the difference between instanceof and getClass()?

Reply:
instanceof use when we use extends